### PR TITLE
Add reminder to check for missing foreign keys

### PR DESCRIPTION
**Pull request by yoomlam**
### Description
Add reminder to check for missing foreign keys in the PR template.
Add `immigrant` gem.
Add notes to `immigrant.rb` to add foreign keys.

### Testing Plan
Test `make check-fks`

### CODE REVIEW

File: lib/helpers/association_wrapper.rb [hunk 1]
```diff
@@ -64,4 +64,13 @@ def grouped_fieldnames_of_typed_associations_with(known_classes)
...
   end
+
+  # To-do: bootstrap Jailer schema-documentation generator with polymorphic associations
+  # clazz=VACOLS::Case
+  # ag=AssocationWrapper.new(clazz).belongs_to.associations.group_by(&:class_name);
+  # ag.transform_values { |assocs| assocs.map{|assoc| [assoc.foreign_key, assoc.foreign_type] } }
+  # def to_jailer_association_csv
+  #   assocs = belongs_to.associations.transform_values { |assocs| assocs.map{|assoc| [assoc.foreign_key, assoc.foreign_type] } }
+  #   ...
+  # end
```
Comment (yoomlam) [inline lib/helpers/association_wrapper.rb:75]: Notes to self

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -29,15 +29,16 @@ Please explain the changes you made here.
...
 * [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
 * [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
-* [ ] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
-* [ ] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
-* [ ] DB schema docs updated with `make docs` (after running `make migrate`)
-* [ ] #appeals-schema notified with summary and link to this PR
-* [ ] Any non-obvious semantics or logic useful for interpreting database data is documented at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)
+* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
+* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
+* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb`
```
Comment (yoomlam) [inline PULL_REQUEST_TEMPLATE.md:38]: Add line 38. All other changes are wording changes to make the list more consistent, e.g., starting with a verb.

File: config/initializers/immigrant.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  { from_table: "board_grant_effectuations", column: "appeal_id" },
+  { from_table: "board_grant_effectuations", column: "decision_document_id" },
+  { from_table: "board_grant_effectuations", column: "end_product_establishment_id" },
+  { from_table: "board_grant_effectuations", column: "granted_decision_issue_id" },
+
+  { from_table: "cached_appeal_attributes", column: "vacols_id", },
+
+  { from_table: "certification_cancellations", column: "certification_id" },
+  { from_table: "claim_establishments", column: "task_id" },
```
Comment (yoomlam) [inline config/initializers/immigrant.rb:23]: @wdrougas-tista will investigate these. If a FK is not added, then add a comment explaining why it wasn't added.

**Review by wdrougas-tista — APPROVED** [review]
✅

**Pull request merged**